### PR TITLE
String slice doc improvements.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/liballoc/str.rs [hunk 1]
```diff
@@ -328,11 +335,16 @@ impl str {
...
     ///
     /// ```
-    /// let v = "🗻∈🌏";
-    /// assert_eq!(Some("🗻"), v.get(0..4));
-    /// assert!(v.get(1..).is_none());
-    /// assert!(v.get(..8).is_none());
-    /// assert!(v.get(..42).is_none());
+    /// let mut v = String::from("🗻∈🌏");
+    ///
+    /// assert_eq!(Some("🗻"), v.get(0..4);
```
Comment (QuietMisdreavus) [inline src/liballoc/str.rs]: Missing a close paren. This is what failed travis.

Comment (frewsxcv) [inline src/liballoc/str.rs]: addressed in the latest force push

**Comment by kennytm** [comment]
@bors retry #40240 🤔. `sccache` error while building LLVM on [`dist-s390x-linux`](https://travis-ci.org/rust-lang/rust/jobs/261331497). Spurious.

```
[00:06:02] sccache: encountered fatal error
[00:06:02] sccache: error : Invalid checksum
[00:06:02] sccache:  cause: Invalid checksum
[00:06:02] make[2]: *** [lib/Target/ARM/CMakeFiles/LLVMARMCodeGen.dir/ARMFastISel.cpp.o] Error 254
[00:06:02] make[2]: *** Deleting file `lib/Target/ARM/CMakeFiles/LLVMARMCodeGen.dir/ARMFastISel.cpp.o'
[00:06:02] make[2]: *** Waiting for unfinished jobs....
...
$ cat obj/tmp/sccache.log
WARN:sccache::cache::s3: Got AWS error: Error(BadHTTPStatus(NotFound), State { next_error: None })
WARN:sccache::cache::s3: Got AWS error: Error(BadHTTPStatus(NotFound), State { next_error: None })
ERROR:sccache::server: ["ARMFastISel.cpp.o"] fatal error: Error(Io(Error { repr: Custom(Custom { kind: Other, error: StringError("Invalid checksum") }) }), State { next_error: None })
ERROR:sccache::server: [Error(Io(Error { repr: Custom(Custom { kind: Other, error: StringError("Invalid checksum") }) }), State { next_error: None })] 	ARMFastISel.cpp.o
WARN:sccache::cache::s3: Got AWS error: Error(BadHTTPStatus(InternalServerError), State { next_error: None })
$ cat /tmp/sccache.log
cat: /tmp/sccache.log: No such file or directory
```

**Pull request merged**